lib/shopify_app/middleware/app_bridge_middleware.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+module ShopifyApp
+  class AppBridgeMiddleware
+    def initialize(app)
+      @app = app
+    end
+
+    def call(env)
+      request = Rack::Request.new(env)
+
+      if request.params.has_key?("shop") && !request.params.has_key?("host")
+        shop = request.params["shop"]
+        host = Base64.urlsafe_encode64("#{shop}/admin", padding: false)
+        request.update_param("host", host)
```

Comment:
From my understanding, the host parameter is subject to change; it's not _always_ going to be the shop Base64 encoded.